### PR TITLE
Update to 0.5.1

### PR DESCRIPTION
**Pull request by dbnicholson**
This is a fake rebase PR on top of the stretch branch. It's intended for a new bionic branch since the new version can't build on stretch. There are a couple new commits at the end to handle building the new RPC code as well as a revert for code that fails with Go < 1.11.

https://phabricator.endlessm.com/T25314

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -44,7 +44,7 @@ else
 endif
 
 protoc:
-	go get -u github.com/golang/protobuf/protoc-gen-go
+	which protoc-gen-go 2>/dev/null || go get -u github.com/golang/protobuf/protoc-gen-go
```
Comment (wjt) [inline Makefile:47]: Seems upstreamable

File: autogen.sh [hunk 2]
```diff
@@ -1,2 +1,5 @@
 #!/bin/sh
-echo 'USE_TEMP_GOPATH = 1' >Makefile.inc
+cat >Makefile.inc <<EOF
+USE_TEMP_GOPATH = 1
+export GOPATH = /usr/share/gocode
```
Comment (wjt) [inline autogen.sh:4]: Yes… sorry about this mess. I also went and dug into how Debian normally packages Go stuff, and even started packaging up mirrorbits' dependencies but then hit a brick wall, and in the end just wanted to get something working.

**Review by wjt — APPROVED** [review]
LGTM

File: Makefile (hunk 1 above)
Comment (dbnicholson) [reply Makefile:47]: Yeah, I'll file a PR when I get a chance.

File: autogen.sh (hunk 2 above)
Comment (dbnicholson) [reply autogen.sh:4]: Nah, I think you did the right thing. I think the forced policy that "everything is in a tree in a top level src directory blah blah blah" is stupid.

**Comment by dbnicholson** [comment]
This got pushed to the bionic branch.

**Pull request closed without merging**